packages/oid4vci/src/metadata/credential-issuer/credential-issuer-metadata.ts

```diff
@@ -13,20 +16,70 @@ import {
 
 const wellKnownCredentialIssuerSuffix = '.well-known/openid-credential-issuer'
 
+type FetchCredentialIssuerMetadataOptions = {
+  callbackContext: Pick<CallbackContext, 'fetch' | 'verifyJwt' | 'signJwt'>
+}
+
 /**
  * @inheritdoc {@link fetchWellKnownMetadata}
  */
 export async function fetchCredentialIssuerMetadata(
   credentialIssuer: string,
-  fetch?: Fetch
+  options: FetchCredentialIssuerMetadataOptions
 ): Promise<{ credentialIssuerMetadata: CredentialIssuerMetadata; originalDraftVersion: Oid4vciDraftVersion } | null> {
-  const wellKnownMetadataUrl = joinUriParts(credentialIssuer, [wellKnownCredentialIssuerSuffix])
-  const result = await fetchWellKnownMetadata(wellKnownMetadataUrl, vCredentialIssuerMetadataWithDraftVersion, fetch)
+  let result: v.InferOutput<typeof vCredentialIssuerMetadataWithDraftVersion> | null = null
+
+  const entityConfiguration = await fetchEntityConfiguration({
```

Comment:
I've been thinking for a while whether the approach is nice and simple, or that it misses a few things.

What i'm not sure about:
- we lose knowledge about whether federation is used
- we lose knowledge on the verification result
- metadata is dependent on the trust chain. We currently don't support metadata policies but since we will in the future i think it might be better to handle this in credo.